retrieval/scrape.go

```diff
@@ -446,6 +461,28 @@ func (sl *scrapeLoop) stop() {
 	<-sl.done
 }
 
+func (sl *scrapeLoop) processScrapeResult(samples model.Samples, scrapeErr error, start time.Time) error {
+	// Collect samples post-relabelling and label handling in a buffer.
+	buf := &bufferAppender{buffer: make(model.Samples, 0, len(samples))}
```

Comment:
"Total allocated" taken from memstats? (The "total" you get from pprof is the total amount of currently allocated heap. The counter is only in memstats (go_memstats_alloc_bytes_total in Prometheus).

Our busiest ingester at SC ingests about 100k samples/s. It allocates ~230MiB/s. The 0.5% added allocation volume sounds about right with these numbers. So perhaps you measured that correctly. (Would be much easier to verify if you reported the method of measurement precisely.)

We still don't know about heap fragmentation problems, and I'm still thinking we shouldn't deliberately perform these additional allocations if we can avoid them as easily as here.